clippy_lints/src/redundant_closure_call.rs

```diff
@@ -155,14 +157,15 @@ impl<'tcx> LateLintPass<'tcx> for RedundantClosureCall {
                         let mut applicability = Applicability::MachineApplicable;
                         let mut hint = Sugg::hir_with_context(cx, body, full_expr.span.ctxt(), "..", &mut applicability);
 
-                        if generator_kind.is_async()
-                            && let hir::ExprKind::Closure(closure) = body.kind
-                        {
-                            let async_closure_body = cx.tcx.hir().body(closure.body);
+                        if generator_kind.is_async() {
+                            // does the innermost closure returns another closure?
+                            if let hir::ExprKind::Closure(closure) = body.kind {
+                                let async_closure_body = cx.tcx.hir().body(closure.body);
 
-                            // `async x` is a syntax error, so it becomes `async { x }`
-                            if !matches!(async_closure_body.value.kind, hir::ExprKind::Block(_, _)) {
-                                hint = hint.blockify();
+                                // `async x` is a syntax error, so it becomes `async { x }`
+                                if !matches!(async_closure_body.value.kind, hir::ExprKind::Block(_, _)) {
+                                    hint = hint.blockify();
+                                }
```

Comment:
Does this part of the diff functionally change anything? As far as I can tell, only indendation changed and the closure check was moved into its own `if`, right? Is that necessary? I believe this isn't technically right or wrong, because this check always succeeds if `generator_kind` is async (this check was already done in `is_async_closure`), but afaict it does it here again to get the inner closure body to figure out if we have `async || 1`, in which case it needs to add a block: `async { 1 }`, or if the body is already a block: `async || { 1 }`, in which case calling `blockify` is unnecessary (would result in `async { { 1 } }`